.tasks/docs.yml

```
@@ -0,0 +1,21 @@
+version: '3'
+
+vars:
+  GOBIN: "{{.USER_WORKING_DIR}}/tools"
+  TFPLUGINDOCS_CMD: "{{.GOBIN}}/tfplugindocs"
+
+tasks:
+  install:
+    desc: "Install documentation tools"
+    cmds:
+      - echo "Installing tfplugindocs..."
+      - mkdir -p {{.GOBIN}}
+      - GOBIN={{.GOBIN}} go install github.com/hashicorp/terraform-plugin-docs/cmd/tfplugindocs@latest
+      - echo "tfplugindocs installed successfully to {{.GOBIN}}"
+
+  default:
+    desc: "Generate provider documentation"
+    cmds:
+      - echo "Generating provider documentation..."
+      - "{{.TFPLUGINDOCS_CMD}}"
+      - echo "Provider documentation generated successfully"
```

NIT: Add a dependency for the documentation tool installation, so it does not fail when a new contributor runs `task docs:default` for the first time:

```suggestion
  default:
    desc: "Generate provider documentation"
    deps:
      - install
    cmds:
      - echo "Generating provider documentation..."
      - "{{.TFPLUGINDOCS_CMD}}"
      - echo "Provider documentation generated successfully"

```

I bumped into:
```
task docs:default
task: [docs:default] echo "Generating provider documentation..."
Generating provider documentation...
task: [docs:default] /Users/rafalkorepta/workspace/redpanda/terraform-provider-redpanda/tools/tfplugindocs
stat /Users/rafalkorepta/workspace/redpanda/terraform-provider-redpanda/tools/tfplugindocs: no such file or directory
task: Failed to run task "docs:default": exit status 127
```